src/pages/SingleCoursePages.js

```diff
@@ -0,0 +1,49 @@
+import React, {useEffect} from "react";
+import { Routes, Route, Link, NavLink, useNavigate, useParams } from 'react-router-dom'
+import apiUtil from '../utils/apiUtil'
+
+//replcae this with the API call. (Finsished)
+//const course = require('./data/courseData.json') 
+
+//localhost:3000/homepage/course/1
+
+//This function is going to be used to display the single course page from when the
+//user selects the course.  
+function DisplayCoursePage(){
+//using useParams() here to grab the number in the course they select then finding the data
+//associated with that course.
+let { paramID } = useParams()
+
+useEffect(() => {
+    const getCourseData = async () => {
+        const response = await apiUtil('get', `/courses/${paramID}`)
+
+        if (response.status === 200) {
+            return (
+                <div>
+                    <h2>{course[paramID].course_name}</h2>
+                    <p>{course[paramID].crn}</p>
+                    <p>{course[paramID].course_instructor}</p>
+                </div>
+            )
+        }
+
+        else {
+            console.log(response.data.error)
+        }
+    }
+    getCourseData()
+
+}, [] )
+
+
+return(
```

Comment:
I think we need a little more than this to be rendered on the page. We definitely want to have the title of the course rendered somewhere, along with its description. And then I imagined we'd also want the lectures displayed (I might be forgetting what else. Check what info we have on Postman like I mentioned above 😄 ). And then this logic would definitely have to condition on whether the current user is a student or teacher in the course. For example, a teacher would want to have the ability to edit the course, access the students enrolled in the course, create a lecture, create a section, access a section, access a lecture, etc. On the other hand, a student would probably just want to access the lectures and not much more.